### ISSUE TITLE
调用基金并预测增幅中报错

### ISSUE THREAD

**Issue by pvfcu3dx** [waditu/tushare#1780]
![Image](https://github.com/user-attachments/assets/1ad2222b-a528-4461-8b6d-425e57876220)

![Image](https://github.com/user-attachments/assets/8f2a9f51-fd9c-4f4b-bf30-2dc98006d431)
使用夸克AI编程生成的代码，其作用是获取某个基金的数据并预测增幅。报错是：np.ndarray没有iloc这个属性。

代码：
import tushare as ts
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression

# 设置 Tushare Token（需要注册并获取）
ts.set_token('053d532360367667aefb77ed907da16be0cbf69aa6b6cf12fe668b89')
pro = ts.pro_api()

# 获取基金数据（这里以富国恒生港股通医疗保健ETF发起式联接C为例）
fund_code = '012793.OF'  # 富国恒生港股通医疗保健ETF发起式联接C
data = pro.fund_nav(ts_code=fund_code, start_date='20180101', end_date='20230101')

# 数据准备
data['nav_date'] = pd.to_datetime(data['nav_date'])
data['nav_date_ordinal'] = data['nav_date'].map(pd.Timestamp.toordinal)

X = data[['nav_date_ordinal']]
y = data['unit_nav'].astype(float)

# 划分训练集和测试集
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

# 训练线性回归模型
model = LinearRegression()
model.fit(X_train, y_train)

# 预测未来的价格
future_dates = pd.date_range(start=data['nav_date'].iloc[-1], periods=365).to_pydatetime().tolist()
future_dates_ordinal = [pd.Timestamp(date).toordinal() for date in future_dates]

future_prices = model.predict(np.array(future_dates_ordinal).reshape(-1, 1))

# 计算历史涨幅
data['daily_return'] = data['unit_nav'].pct_change()

# 计算预测涨幅
future_returns = (future_prices / np.roll(future_prices, 1)) - 1
future_returns.iloc[0] = 0  # 第一天的涨幅设为0

# 绘制结果
plt.figure(figsize=(14, 7))
plt.plot(data['nav_date'], data['unit_nav'], label='Historical NAV')
plt.plot(future_dates, future_prices, linestyle='--', color='red', label='Predicted NAV')
plt.xlabel('Date')
plt.ylabel('Net Asset Value (NAV)')
plt.title(f'{fund_code} Fund NAV Prediction')
plt.legend()
plt.show()

# 绘制涨幅结果
plt.figure(figsize=(14, 7))
plt.plot(data['nav_date'][1:], data['daily_return'][1:], label='Historical Daily Return')
plt.plot(future_dates[1:], future_returns[1:], linestyle='--', color='green', label='Predicted Daily Return')
plt.xlabel('Date')
plt.ylabel('Daily Return')
plt.title(f'{fund_code} Fund Daily Return Prediction')
plt.legend()
plt.show()
